presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileFunctions.java

```diff
@@ -371,16 +394,101 @@ private static Point tileXYToLatitudeLongitude(int tileX, int tileY, int zoomLev
         return new Point(longitude, latitude);
     }
 
-    private static BingTile latitudeLongitudeToTile(double latitude, double longitude, int zoomLevel)
+    /**
+     * Given latitude and longitude in degrees, and the level of detail, the pixel XY coordinates can be calculated as follows:
+     * sinLatitude = sin(latitude * pi/180)
+     * pixelX = ((longitude + 180) / 360) * 256 * 2level
+     * pixelY = (0.5 – log((1 + sinLatitude) / (1 – sinLatitude)) / (4 * pi)) * 256 * 2level
+     * The latitude and longitude are assumed to be on the WGS 84 datum. Even though Bing Maps uses a spherical projection,
+     * it’s important to convert all geographic coordinates into a common datum, and WGS 84 was chosen to be that datum.
+     * The longitude is assumed to range from -180 to +180 degrees, and the latitude must be clipped to range from -85.05112878 to 85.05112878.
+     * This avoids a singularity at the poles, and it causes the projected map to be square.
+     * <p>
+     * reference: https://msdn.microsoft.com/en-us/library/bb259689.aspx
+     */
+    private static int longitudeToTileX(double longitude, int mapSize)
     {
         double x = (longitude + 180) / 360;
+        return axisToCoordinates(x, mapSize);
+    }
+
+    private static int longitudeToTileY(double latitude, int mapSize)
+    {
         double sinLatitude = Math.sin(latitude * Math.PI / 180);
         double y = 0.5 - Math.log((1 + sinLatitude) / (1 - sinLatitude)) / (4 * Math.PI);
+        return axisToCoordinates(y, mapSize);
+    }
 
+    /**
+     * Take axis and convert it to Tile coordinates
+     *
+     * @param axis
+     * @param mapSize
+     * @return
+     */
+    private static int axisToCoordinates(double axis, int mapSize)
+    {
+        int tileAxis = (int) clip(axis * mapSize, 0, mapSize - 1);
+        return tileAxis / TILE_PIXELS;
+    }
+
+    private static boolean coordinateInsideBoundary(int coordinate, int mapSize)
+    {
+        return (coordinate >= 0 && coordinate <= (mapSize - 1) / TILE_PIXELS);
+    }
+
+    /**
+     * Returns a Bing tile at a given zoom level containing a point at a given latitude and longitude.
+     * Latitude must be within [-85.05112878, 85.05112878] range. Longitude must be within [-180, 180] range.
+     * Zoom levels from 1 to 23 are supported.
+     *
+     * @param latitude
+     * @param longitude
+     * @param zoomLevel
+     * @return
+     */
+    private static BingTile latitudeLongitudeToTile(double latitude, double longitude, int zoomLevel)
```

Comment:
Move it up. Prefer putting methods that are used by other method below the point of usage.

Reply:
Just to make it more clear

If you have 3 methods. `a()`, `b()` , `c()`. And `c()` is used within `b()` and `b()` is used within `a()`. So in file they should come in order you reach them when reading the code `a()`  (`a()` -> `b()` -> `c()`) order

Reply:
Will do. 

Though, usually I group the methods of similar kind together, if within a class. People use IDEs to jump between method anyways, I doubt the order will make much difference.

Reply:
> Though, usually I group the methods of similar kind together, if within a class.

This is not a hard rule, but rather a guideline. Sometimes it makes more sense to group them by some other common property.

> People use IDEs to jump between method anyways

We spend a lot of time reading diffs. You cannot just so easy on a `Github`. It makes somehow easier to read when you have at leas some kind of order (maybe not a perfect though).